controller/controller.go

```diff
@@ -705,7 +705,13 @@ func NewProvisionController(
 	// PersistentVolumes
 
 	volumeHandler := cache.ResourceEventHandlerFuncs{
-		AddFunc:    func(obj interface{}) { controller.enqueueVolume(obj) },
+		AddFunc: func(obj interface{}) {
+			if err = controller.volumes.Add(obj); err != nil {
```

Comment:
The volume coming from informer already is in the informer cache, you don't need to add it again.

What you need to do is to add the volume to the store *after* the provisioner saved the volume to the API server here
https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/904215ea993ad0afd415a063ab80d5711ba9802e/controller/volume_store.go#L157
and here
https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/904215ea993ad0afd415a063ab80d5711ba9802e/controller/volume_store.go#L215

Reply:
I actually moved it here https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/controller/controller.go#L1440
because 'volumes' is a member of 'ProvisionController`